### PR TITLE
Add --insecure-tls flag to disable SSL cert verification

### LINKED ISSUE #2656: Add a flag to skip TLS verification for self-signed certificates
Output of `restic version`
--------------------------

restic 0.9.6

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------

It would be helpful if Restic had a flag to skip TLS verification for self-signed certificates, as an alternative to passing a certificate bundle via `--cacert`.

What are you trying to do?
--------------------------

Velero supports a `--insecure-skip-tls-verify` flag that allows turning off TLS verification for self-signed certificates when talking to storage backends which is handy for development purposes. However, that doesn't work with Velero's Restic integration for backing up Kubernetes volumes, since Restic doesn't support an equivalent flag.

I'll put up a PR with a possible implementation shortly.

### PR DESCRIPTION
**Pull request by mansam**
What is the purpose of this change? What does it change?
--------------------------------------------------------

This adds a flag to disable TLS certificate verification for self-signed certificates as described in https://github.com/restic/restic/issues/2656.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

Closes https://github.com/restic/restic/issues/2656

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [ ] I have added tests for all changes in this PR
- [x] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

**Comment by apollotonkosmo** [comment]
@mansam this is indeed very useful. I used your repo/branch on my machine and it works. Why isn't this still merged?

**Comment by ofinto** [comment]
I agree with @apollotonkosmo, this is a very useful feature. I tried to use restic with self-signed certs and for that I needed to create PKI with CA, that's more difficult than ignoring ssl. Many people can use restic in privat network with minio.

File: cmd/restic/global.go [hunk 1]
```diff
@@ -105,6 +106,7 @@ func init() {
 	f.BoolVar(&globalOptions.NoCache, "no-cache", false, "do not use a local cache")
 	f.StringSliceVar(&globalOptions.CACerts, "cacert", nil, "`file` to load root certificates from (default: use system certificates)")
 	f.StringVar(&globalOptions.TLSClientCert, "tls-client-cert", "", "path to a file containing PEM encoded TLS client certificate and private key")
+	f.BoolVar(&globalOptions.InsecureSkipTLSVerify, "insecure-skip-tls-verify", false, "skip TLS certificate verification when connecting to the repo (insecure)")
```
Comment (MichaelEischer) [inline cmd/restic/global.go:109]: Please rename the flag to `--insecure-tls`. Although technically correct, the current name is rather hard to parse.

File: changelog/unreleased/issue-2656 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Enhancement: Add flag to disable TLS verification for self-signed certificates
+
+We've added a flag, `--insecure-skip-tls-verify`, to allow disabling
+TLS verification for self-signed certificates in order to support
+some development workflows.
+
+https://github.com/restic/restic/issues/2656
```
Comment (MichaelEischer) [inline changelog/unreleased/issue-2656:7]: Please also add a link to this PR.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]

**Comment by MichaelEischer** [comment]
@mansam Are you still interested in getting this PR merged? I wanted to make the cleanups myself, but wasn't allowed to push to your branch.

**Comment by mansam** [comment]
@MichaelEischer Apologies for the delay in responding to this. I am, I'll update the PR shortly.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks for the cleanup!

**Review by lucasmlousada-scality — COMMENTED** [review]
Thanks :)
